test/components/project/home-block.spec.js

```diff
@@ -131,4 +133,51 @@ describe('ProjectHomeBlock', () => {
     const rows = block.findAllComponents(FormRow);
     rows.map((row) => row.props().form.name).should.eql(['a', 'c', 'd', 'e']);
   });
+
+  it('shows the correct number of forms and datasets if there are only a few', () => {
+    testData.extendedProjects.createPast(1);
+    testData.extendedForms.createPast(3);
+    testData.extendedDatasets.createPast(3);
+    const block = mountComponent();
+    block.findAllComponents(FormRow).length.should.equal(3);
+    block.findAllComponents(DatasetRow).length.should.equal(3);
+    block.find('.expand-button').exists().should.be.false();
+    block.find('.margin').exists().should.be.true();
+  });
+
+  it('shows the correct number of datasets if there are a lot and some should be hidden', () => {
+    testData.extendedProjects.createPast(1);
+    testData.extendedDatasets.createPast(4);
+    const block = mountComponent();
+    block.findAllComponents(DatasetRow).length.should.equal(3);
+    const expand = block.find('.expand-button');
```

Comment:
there are potentially 2 expand buttons, do you want to make sure it's the dataset one? and check that the form one doesn't exist?